mk-configure: update to 0.37.0.

mobinmob opened the pull request:
<!-- Mark items with [x] where applicable -->

#### General
- [ ] This is a new package and it conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements)

#### Have the results of the proposed changes been tested?
- [ ] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me
- [X] I generally don't use the affected packages but briefly tested this PR

<!--
If GitHub CI cannot be used to validate the build result (for example, if the
build is likely to take several hours), make sure to
[skip CI](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#continuous-integration).
When skipping CI, uncomment and fill out the following section.
Note: for builds that are likely to complete in less than 2 hours, it is not
acceptable to skip CI.
-->
<!-- 
#### Does it build and run successfully? 
(Please choose at least one native build and, if supported, at least one cross build. More are better.)
- [ ] I built this PR locally for my native architecture, (ARCH-LIBC)
- [ ] I built this PR locally for these architectures (if supported. mark crossbuilds):
  - [ ] aarch64-musl
  - [ ] armv7l
  - [ ] armv6l-musl
-->

ericonr commented:
Clean the musl specific part from `pre_check`, it's no longer valid.

The `bmake` error failure seems to be due to an extra `bmake` error message?

```
~/v/void-packages master [~5] 😨 31s ➜ cat masterdir/builddir/mk-configure-mk-configure-0.36.0/tests/test_subprj_dash/expect.out
Dash symbol is not allowed inside subdir (another-one prog100-500)
~/v/void-packages master [~5] 😨 ➜ cat masterdir/builddir/mk-configure-mk-configure-0.36.0/tests/test_subprj_dash/test_subprj_dash.test.out

Dash symbol is not allowed inside subdir (another-one prog100-500)
bmake: stopped in /objdir
```

mobinmob commented:
> Clean the musl specific part from `pre_check`, it's no longer valid.
> 
> The `bmake` error failure seems to be due to an extra `bmake` error message? (It happens on the previous version as well)
> 
> ```
> ~/v/void-packages master [~5] 😨 31s ➜ cat masterdir/builddir/mk-configure-mk-configure-0.36.0/tests/test_subprj_dash/expect.out
> Dash symbol is not allowed inside subdir (another-one prog100-500)
> ~/v/void-packages master [~5] 😨 ➜ cat masterdir/builddir/mk-configure-mk-configure-0.36.0/tests/test_subprj_dash/test_subprj_dash.test.out
> 
> Dash symbol is not allowed inside subdir (another-one prog100-500)
> bmake: stopped in /objdir
> ```

Yes, different output than expected. Removed the test and others that depend on it and added libbsd-devel to checkdepeds. It should work now - it is fine locally.

mobinmob commented:
Added musl-fts-devel to checkdepends on musl.
